### ISSUE TITLE
Viewpoint Test Checker: `@PolyVP` method lets a `@B` value be stored in an `@A` field

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2074]
The Viewpoint Test Checker reports no errors on this file:

```java
import viewpointtest.quals.*;

public class PolyRDQUnsound {
    @ReceiverDependentQual Object f;

    static void set(@PolyVP PolyRDQUnsound c, @PolyVP Object o) {
        c.f = o;
    }

    static void test(@A PolyRDQUnsound a, @B Object b) {
        set(a, b);
        @A Object aObj = a.f;
    }
}
```

After `set(a, b)`, `a.f` holds a `@B` value, but it is read as `@A` without any error.

Writing the same thing directly is rejected. Through a `@Top` receiver (as in `VPAExamples`), `top.f = b` gives `assignment.type.incompatible`.

The expected result is an error at `c.f = o` in `set`.
